core/src/test/scala/akka/persistence/cassandra/EventsByTagRestartSpec.scala

```diff
@@ -83,5 +84,43 @@ class EventsByTagRestartSpec extends CassandraSpec(EventsByTagRestartSpec.config
       tagProbe.expectNoMessage(waitTime)
       tagProbe.cancel()
     }
+
+    "use correct tag sequence nrs when actor is stopped prematurely" in {
+      val queryJournal = PersistenceQuery(system).readJournalFor[CassandraReadJournal](CassandraReadJournal.Identifier)
+      val p2 = system.actorOf(TestTaggingActor.props("p2", Set("green")))
+      val probe1 = TestProbe()
+      probe1.watch(p2)
+      p2.tell("e1", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e2", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e3", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e4", probe1.ref)
+      // the PoisonPill will cause the actor to be stopped before the journal write has been completed,
+      // and is therefore similar to if the CircuitBreaker would trigger
+      p2 ! PoisonPill
+      probe1.expectTerminated(p2)
+
+      Thread.sleep(2000)
+      val probe2 = TestProbe()
+      // tag will be missing until we restart actor, (is that ok?)
+      val p2b = system.actorOf(TestTaggingActor.props("p2", Set("green")))
+      p2b.tell("e5", probe2.ref)
+      probe2.expectMsg(Ack)
+      p2b.tell("e6", probe2.ref)
+      probe2.expectMsg(Ack)
+
+      val greenTags = queryJournal.eventsByTag(tag = "green", offset = NoOffset)
+      val tagProbe = greenTags.runWith(TestSink.probe[Any](system))
+      tagProbe.request(10)
+      (1 to 6).foreach { n =>
+        val event = s"e$n"
+        system.log.debug("Expecting event {} sequenceNr {}", event, n)
+        tagProbe.expectNextPF { case EventEnvelope(_, "p2", `n`, `event`) => }
+      }
```

Comment:
without the fix this would not complete, because it found "duplicate/wrong" seqNr 1 in the eventsByTag query and wouldn't deliver that event (as expected)

Reply:
I'll add a comment clarifying that. Always good with context in a year when we look at this again.